### PR TITLE
Release 0.4.1-0.1.0

### PR DESCRIPTION
**Pull request by richerve**
Cut first release of the plugin on kanopy-platform.

## Release process

- Merge this PR to master
- Tag master with v0.1.0 and push tag to origin
- Use goreleaser in local machine to generate artifacts
- Create a Release in github for the v0.1.0 tag and attach the artifacts generated by goreleaser.

Goreleaser can push the artifacts to github directly but I think for this one of release is not necessary to create a token to do that.

Here's a snaphot of what goreleaser will do. Replacing `0.0.0-SNAPSHOT-7431bf6` with `0.4.1-0.1.0` after the git tag is created.

```
❱ goreleaser release --snapshot --clean
  • starting release...
  • loading                                          path=.goreleaser.yml
  • skipping announce, publish and validate...
  • loading environment variables
  • getting and validating git state
    • ignoring errors because this is a snapshot     error=git doesn't contain any tags. Either add a tag or use --snapshot
    • git state                                      commit=7431bf6dc493c07a32e997c4c79fdb1cdf76c622 branch=main current_tag=v0.0.0 previous_tag=<unknown> dirty=false
    • pipe skipped                                   reason=disabled during snapshot mode
  • parsing tag
  • setting defaults
  • snapshotting
    • building snapshot...                           version=0.0.0-SNAPSHOT-7431bf6
  • checking distribution directory
  • loading go mod information
  • build prerequisites
  • writing effective config file
    • writing                                        config=dist/config.yaml
  • building binaries
    • building                                       binary=dist/helm-mapkubeapis_windows_arm64/mapkubeapis.exe
    • building                                       binary=dist/helm-mapkubeapis_linux_amd64_v1/mapkubeapis
    • building                                       binary=dist/helm-mapkubeapis_darwin_amd64_v1/mapkubeapis
    • building                                       binary=dist/helm-mapkubeapis_darwin_arm64/mapkubeapis
    • building                                       binary=dist/helm-mapkubeapis_linux_arm64/mapkubeapis
    • building                                       binary=dist/helm-mapkubeapis_windows_amd64_v1/mapkubeapis.exe
    • took: 56s
  • archives
    • creating                                       archive=dist/helm-mapkubeapis_0.0.0-SNAPSHOT-7431bf6_darwin_amd64.tar.gz
    • creating                                       archive=dist/helm-mapkubeapis_0.0.0-SNAPSHOT-7431bf6_windows_amd64.tar.gz
    • creating                                       archive=dist/helm-mapkubeapis_0.0.0-SNAPSHOT-7431bf6_windows_arm64.tar.gz
    • creating                                       archive=dist/helm-mapkubeapis_0.0.0-SNAPSHOT-7431bf6_linux_arm64.tar.gz
    • creating                                       archive=dist/helm-mapkubeapis_0.0.0-SNAPSHOT-7431bf6_linux_amd64.tar.gz
    • creating                                       archive=dist/helm-mapkubeapis_0.0.0-SNAPSHOT-7431bf6_darwin_arm64.tar.gz
    • took: 4s
  • calculating checksums
  • storing release metadata
    • writing                                        file=dist/artifacts.json
    • writing                                        file=dist/metadata.json
  • release succeeded after 59s
  • thanks for using goreleaser!
```

### CODE REVIEW

File: plugin.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: "mapkubeapis"
-version: "0.4.1"
+version: "v0.1.0"
```
Comment (yuzhouliu9) [inline plugin.yaml]: I think we should use `0.4.1-0.1.0` to indicate we've made changes on top of 0.4.1. Easier for us to know how behind we are from upstream

@colinhoglund is that the way?

**Comment by yuzhouliu9** [comment]
I've never used goreleaser before, but from the [quick start](https://goreleaser.com/quick-start/) it seems a bit easier than your description, when using your own token. I think it's just

1. Locally `git tag -a <tag> -m "First release"`, `git push origin <tag>`
2. `export GITHUB_TOKEN="YOUR_GH_TOKEN"` making sure the [token](https://github.com/settings/tokens) has `write:packages` permissions, and authorized for `kanopy-platform` org
3. At root of repo: `goreleaser release`

File: plugin.yaml (hunk 1 above)
Comment (colinhoglund) [reply plugin.yaml]: 👍 [This](https://github.com/kanopy-platform/drone/releases/tag/v2.17.0-0.1.0) is the way.

Comment (richerve) [reply plugin.yaml]: ok, will do. I figured that we are essentially doing a hard fork so can keep our own versioning.
Another thing I wasn't sure is if the helm plugin would support the hyphen annotation but I think it should as it indicates it reads a SemVer version.

**Comment by richerve** [comment]
> I've never used goreleaser before, but from the [quick start](https://goreleaser.com/quick-start/) it seems a bit easier than your description, when using your own token. I think it's just
> 
>     1. Locally `git tag -a <tag> -m "First release"`, `git push origin <tag>`
> 
>     2. `export GITHUB_TOKEN="YOUR_GH_TOKEN"` making sure the [token](https://github.com/settings/tokens) has `write:packages` permissions, and authorized for `kanopy-platform` org
> 
>     3. At root of repo: `goreleaser release`

Right, I suppose I can use my token to deploy too

**Review by rojanDinc — APPROVED** [review]

**Pull request merged**